test/stacktraces.jl

```diff
@@ -191,3 +191,27 @@ let bt
     end
     @test any(s->startswith(string(s), "f33065(x::Float32, y::Float32; b::Float64, a::String, c::"), bt)
 end
+
+# Opaque closure backtraces
+oc_error_template(x) = error(x)
+let ci = code_lowered(oc_error_template, Tuple{Any})[]
+    # This should run in the interpreter
+    oc = eval(Expr(:new_opaque_closure, Tuple{Any}, false, Any, Any,
+                   Expr(:opaque_closure_method, :oc_method_name, 1, LineNumberNode(1, :foo), ci)))
+    bt = try
+        oc(1)
+    catch
+        stacktrace(catch_backtrace())
+    end
+    @test bt[findfirst(s->startswith(string(s), "oc_method_name"), bt)].linfo !== nothing
```

Comment:
Maybe also add a test that it gets printed correctly (not tested):
```suggestion
    @test bt[findfirst(s->startswith(string(s), "oc_method_name"), bt)].linfo !== nothing
    @test occursin("oc_method_name(x::Any)", sprint(Base.show_backtrace, bt))
```